### PR TITLE
Feat: enhance addon init

### PR DESCRIPTION
**Pull request by charlie0129**
Signed-off-by: Charlie Chiang <charlie_c_0129@outlook.com>

### Description of your changes

- Make blue color clearer on black backgrounds
- Generate sample files by default to get user started (can be skipped)
- Support `--url` to add a ref-objects component (use yaml files from URLs)
- Rename existing `readme` to `README`
- Refactor code structure

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Unit-tests.

Manually tested with several possible combinations of parameters.

### CODE REVIEW

**Comment by wonderflow** [comment]
> Rename existing readme to README

Please make sure API is compatible with this change.

**Comment by charlie0129** [comment]
> Please make sure API is compatible with this change.
 
This should work just fine:
1. Before this change, the existing addons have both `readme.md` and `README.md`. They all worked, meaning changing case should breaking anything.
2. This change should only affect reading the `README.md` from disk. While most filesystems are case-insensitive, so it should read the right file.

File: references/cli/addon_test.go [hunk 1]
```diff
@@ -421,16 +421,16 @@ func TestGenerateParameterString(t *testing.T) {
...
 	cmd.SetArgs([]string{})
 	err := cmd.Execute()
 	assert.ErrorContains(t, err, "required")
 
-	cmd.SetArgs([]string{"--chart", "a", "--helm-repo-url", "https://some.com", "--version", "c"})
+	cmd.SetArgs([]string{"--chart", "a", "--helm-repo", "https://some.com", "--chart-version", "c"})
 	err = cmd.Execute()
 	assert.ErrorContains(t, err, "required")
 
-	cmd.SetArgs([]string{"test-addon", "--chart", "a", "--helm-repo-url", "https://some.com", "--version", "c"})
```
Comment (wonderflow) [inline references/cli/addon_test.go:433]: is this a break change?

**Review by wonderflow — COMMENTED** [review]
does any docs need to change after this PR?

File: references/cli/addon_test.go (hunk 1 above)
Comment (charlie0129) [reply references/cli/addon_test.go:433]: Yes, the parameters are adjusted. But this function is for humans, there are no automation tools or other things that uses this, so it won't affect much.

The CLI examples have been updated, so the user can refer to help.

Comment (wonderflow) [reply references/cli/addon_test.go:433]: well, we're not release it in a formal release, so we can change it for now. But next time, we should keep the consistence

**Review by wangyikewxgm — APPROVED** [review]

**Review by Somefive — APPROVED** [review]

**Pull request merged**